Patch by MSP-Greg
Allows using mkmf.rb when running test-all tests from install
diff --git a/lib/mkmf.rb b/lib/mkmf.rb
index b035670466..191ac20b57 100644
--- a/lib/mkmf.rb
+++ b/lib/mkmf.rb
@@ -240,6 +240,10 @@ def map_dir(dir, map = nil)
   elsif File.exist?(($hdrdir = ($top_srcdir ||= topdir) + "/include")  + "/ruby.h")
     $topdir ||= RbConfig::CONFIG["topdir"]
     $arch_hdrdir = "$(extout)/include/$(arch)"
+  elsif File.exist?(($hdrdir = RbConfig::CONFIG["rubyhdrdir"]) + "/ruby/ruby.h")
+    $extmk = false
+    $topdir = $top_srcdir = $hdrdir
+    $arch_hdrdir = RbConfig::CONFIG["rubyarchhdrdir"]
   else
     abort <<MESSAGE
 mkmf.rb can't find header files for ruby at #{$hdrdir}/ruby.h
@@ -451,7 +455,7 @@ def create_tmpsrc(src)
   def have_devel?
     unless defined? $have_devel
       $have_devel = true
-      $have_devel = try_link(MAIN_DOES_NOTHING)
+      #$have_devel = try_link(MAIN_DOES_NOTHING)
     end
     $have_devel
   end
